[sdl2] build failure on <any>-android with ndk 27

### Operating system

Linux

### Compiler

Clang

### Steps to reproduce the behavior

```Shell
# Download android ndk r27 from https://developer.android.com/ndk/downloads
export ANDROID_NDK_HOME=$HOME/Downloads/android-ndk-r27/
run `vcpkg install sdl2:arm64-android`
```


### Failure logs

```
   [246/266] /usr/local/lib/android/sdk/ndk/27.0.12077973/toolchains/llvm/prebuilt/linux-x86_64/bin/clang --target=aarch64-none-linux-android21 --sysroot=/usr/local/lib/android/sdk/ndk/27.0.12077973/toolchains/llvm/prebuilt/linux-x86_64/sysroot -DGL_GLEXT_PROTOTYPES -DSDL_STATIC_LIB -DUSING_GENERATED_CONFIG_H -I/home/runner/work/openblack/openblack/vcpkg/buildtrees/sdl2/arm64-android-dbg/include -I/home/runner/work/openblack/openblack/vcpkg/buildtrees/sdl2/arm64-android-dbg/include/SDL2 -I/home/runner/work/openblack/openblack/vcpkg/buildtrees/sdl2/arm64-android-dbg/include-config-debug/SDL2 -I/home/runner/work/openblack/openblack/vcpkg/buildtrees/sdl2/arm64-android-dbg/include-config-debug -I/usr/local/lib/android/sdk/ndk/27.0.12077973/sources/android/cpufeatures -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security  -fPIC   -idirafter "/home/runner/work/openblack/openblack/vcpkg/buildtrees/sdl2/src/as
  FAILED: CMakeFiles/SDL2-static.dir/src/sensor/android/SDL_androidsensor.c.o 
  /usr/local/lib/android/sdk/ndk/27.0.12077973/toolchains/llvm/prebuilt/linux-x86_64/bin/clang --target=aarch64-none-linux-android21 --sysroot=/usr/local/lib/android/sdk/ndk/27.0.12077973/toolchains/llvm/prebuilt/linux-x86_64/sysroot -DGL_GLEXT_PROTOTYPES -DSDL_STATIC_LIB -DUSING_GENERATED_CONFIG_H -I/home/runner/work/openblack/openblack/vcpkg/buildtrees/sdl2/arm64-android-dbg/include -I/home/runner/work/openblack/openblack/vcpkg/buildtrees/sdl2/arm64-android-dbg/include/SDL2 -I/home/runner/work/openblack/openblack/vcpkg/buildtrees/sdl2/arm64-android-dbg/include-config-debug/SDL2 -I/home/runner/work/openblack/openblack/vcpkg/buildtrees/sdl2/arm64-android-dbg/include-config-debug -I/usr/local/lib/android/sdk/ndk/27.0.12077973/sources/android/cpufeatures -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security  -fPIC   -idirafter "/home/runner/work/openblack/openblack/vcpkg/buildtrees/sdl2/src/ase-2.30.5-4
  Error: /home/runner/work/openblack/openblack/vcpkg/buildtrees/sdl2/src/ase-2.30.5-4439cd0809.clean/src/sensor/android/SDL_androidsensor.c:164:9: error: 'ALooper_pollAll' is unavailable: obsoleted in Android 1 - ALooper_pollAll may ignore wakes. Use ALooper_pollOnce instead. See The API documentation for more information
    164 |     if (ALooper_pollAll(0, NULL, &events, (void **)&source) == LOOPER_ID_USER) {
        |         ^
  /usr/local/lib/android/sdk/ndk/27.0.12077973/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/android/looper.h:228:5: note: 'ALooper_pollAll' has been explicitly marked unavailable here
    228 | int ALooper_pollAll(int timeoutMillis, int* outFd, int* outEvents, void** outData)
        |     ^
  1 error generated.
```

### Additional context

Issue on SDL2, but this is not yet fixed in the next version released two days ago.
https://github.com/libsdl-org/SDL/issues/9792
The fix is this
https://github.com/libsdl-org/SDL/commit/558630d59c2db73ac7c0c0cae000e4bee6ef42cf

--- a/ports/sdl2/poll-fix.patch
+++ b/ports/sdl2/poll-fix.patch
@@ -1,0 +1,13 @@
+diff --git a/src/sensor/android/SDL_androidsensor.c b/src/sensor/android/SDL_androidsensor.c
+index 3f57253cd..c06faf2ee 100644
+--- a/src/sensor/android/SDL_androidsensor.c
++++ b/src/sensor/android/SDL_androidsensor.c
+@@ -161,7 +161,7 @@ static void SDL_ANDROID_SensorUpdate(SDL_Sensor *sensor)
+     ASensorEvent event;
+     struct android_poll_source *source;
+ 
+-    if (ALooper_pollAll(0, NULL, &events, (void **)&source) == LOOPER_ID_USER) {
++    if (ALooper_pollOnce(0, NULL, &events, (void **)&source) == LOOPER_ID_USER) {
+         SDL_zero(event);
+         while (ASensorEventQueue_getEvents(sensor->hwdata->eventqueue, &event, 1) > 0) {
+             SDL_PrivateSensorUpdate(sensor, 0, event.data, SDL_arraysize(event.data));

--- a/ports/sdl2/portfile.cmake
+++ b/ports/sdl2/portfile.cmake
@@ -8,6 +8,7 @@ vcpkg_from_github(
         deps.patch
         alsa-dep-fix.patch
         cxx-linkage-pkgconfig.diff
+        poll-fix.patch # Remove with next version (2.30.7).
 )
 
 string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "static" SDL_STATIC)

--- a/ports/sdl2/vcpkg.json
+++ b/ports/sdl2/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "sdl2",
   "version": "2.30.6",
+  "port-version": 1,
   "description": "Simple DirectMedia Layer is a cross-platform development library designed to provide low level access to audio, keyboard, mouse, joystick, and graphics hardware via OpenGL and Direct3D.",
   "homepage": "https://www.libsdl.org/download-2.0.php",
   "license": "Zlib",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -8070,7 +8070,7 @@
     },
     "sdl2": {
       "baseline": "2.30.6",
-      "port-version": 0
+      "port-version": 1
     },
     "sdl2-gfx": {
       "baseline": "1.0.4",

--- a/versions/s-/sdl2.json
+++ b/versions/s-/sdl2.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "35851962eb04c90f32822e7574ae69265113d88b",
+      "version": "2.30.6",
+      "port-version": 1
+    },
+    {
       "git-tree": "0ddd2439efece30735786a3530fced0bb645c0af",
       "version": "2.30.6",
       "port-version": 0

